flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveScheduler.java

```diff
@@ -1243,7 +1246,8 @@ public void goToRestarting(
             OperatorCoordinatorHandler operatorCoordinatorHandler,
             Duration backoffTime,
             boolean forcedRestart,
-            List<ExceptionHistoryEntry> failureCollection) {
+            List<ExceptionHistoryEntry> failureCollection,
+            boolean isFromFailure) {
```

Comment:
Do we need a new parameter or can we derive this new parameter from `!failureCollection.isEmpty()`?